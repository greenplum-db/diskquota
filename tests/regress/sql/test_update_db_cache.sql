--start_ignore
CREATE DATABASE test_db_cache;
--end_ignore

\c test_db_cache
CREATE EXTENSION diskquota;
CREATE EXTENSION diskquota_test;

-- Wait until the db cache gets updated 
SELECT diskquota.wait_for_worker_new_epoch();

CREATE TABLE t(i) AS SELECT generate_series(1, 100000)
DISTRIBUTED BY (i);

SELECT diskquota.wait_for_worker_new_epoch();

SELECT tableid::regclass, size, segid
FROM diskquota.table_size
WHERE tableid = 't'::regclass
ORDER BY segid;

DROP TABLE t;
DROP EXTENSION diskquota;

\c contrib_regression
DROP DATABASE test_db_cache;
