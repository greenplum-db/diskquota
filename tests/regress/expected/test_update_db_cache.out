--start_ignore
CREATE DATABASE test_db_cache;
--end_ignore
\c test_db_cache
CREATE EXTENSION diskquota;
CREATE EXTENSION diskquota_test;
-- Wait until the db cache gets updated 
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

CREATE TABLE t(i) AS SELECT generate_series(1, 100000)
DISTRIBUTED BY (i);
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

SELECT tableid::regclass, size, segid
FROM diskquota.table_size
WHERE tableid = 't'::regclass
ORDER BY segid;
 tableid |  size   | segid 
---------+---------+-------
 t       | 3637248 |    -1
 t       | 1212416 |     0
 t       | 1212416 |     1
 t       | 1212416 |     2
(4 rows)

DROP TABLE t;
DROP EXTENSION diskquota;
\c contrib_regression
DROP DATABASE test_db_cache;
